### PR TITLE
plugins: convert: make a bit 'natural'

### PR DESCRIPTION
**Pull request by thewisenerd**

### CODE REVIEW

File: plugins/convert.py [hunk 1]
```diff
@@ -31,17 +31,26 @@ def convert(inp, bot=None):
...
-    except IndexError:
+        parts.remove('TO')
+    except ValueError:
+        pass
+
+    amount = None
+    for x in parts:
+        try:
+            amount = float(x)
+            parts.remove(x)
```
Comment (antdking) [inline plugins/convert.py]: try: except: else: pattern please. 

for: else: pattern please.

``` python
for x in list(parts):
  try:
    amount = float(x)
  except ValueError:
    continue
  parts.remove(x)
  break
else:
  amount = 1.0
```

File: plugins/convert.py [hunk 2]
```diff
@@ -31,17 +31,26 @@ def convert(inp, bot=None):
...
+        except ValueError:
+            pass
+    if (amount == None):
+        amount = 1.0
+
+    # this, or try/except
+    if (len(parts) != 2):
```
Comment (antdking) [inline plugins/convert.py]: No parentheses pleasd

Comment (antdking) [inline plugins/convert.py]: Actually, can you keep the original try except please. Catch the index error, don't care if they've got other stuff written in the arguments

Comment (thewisenerd) [inline plugins/convert.py]: :+1:

**Pull request merged**